rpc/handlers.go

```diff
@@ -734,3 +738,26 @@ func setEventFilterRange(filter *blockchain.EventFilter, fromID, toID *BlockID,
 	}
 	return set(blockchain.EventFilterTo, toID)
 }
+
+// AddInvokeTransaction relays an invoke transaction to the gateway.
+//
+// It follows the specification defined here:
+// https://github.com/starkware-libs/starknet-specs/blob/a789ccc3432c57777beceaa53a34a7ae2f25fda0/api/starknet_write_api.json#L11
+func (h *Handler) AddInvokeTransaction(invokeTx *gateway.BroadcastedInvokeTxn) (*AddInvokeTransactionResponse, *jsonrpc.Error) {
+	if invokeTx.Type == "INVOKE" {
+		invokeTx.Type = "INVOKE_FUNCTION"
+	}
+
+	resp, err := h.gatewayClient.AddInvokeTransaction(context.TODO(), invokeTx)
+	if err != nil {
+		return nil, &jsonrpc.Error{
+			Code:    0,
```

Comment:
Are you sure 0 is the right error code here?

Reply:
it should be non zero, fixing it

Reply:
It's not clear what the error code should be in some cases. For example, if the max fee is too high, the sequencer returns an error code of  `{"code": "StarknetErrorCode.OUT_OF_RANGE_FEE", "message": "Fee 0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa is out of range"}`, whereas pathfinder (via infura endpoint) returns `{"code":-32603,"message":"Sending V1 invoke transaction to gateway"}`. Should we just specify our own error code?

Reply:
Any errors related to validation or incorrect versions should be caught prior to this, so it's mainly errors related to high gas fees, etc.

Reply:
I added a set of error codes, following pathfinders approach.